sensors/sqs_sensor.py

```diff
@@ -87,61 +96,156 @@ def _get_config_entry(self, key, prefix=None):
-                               region_name=self.aws_region)
+    def _setup_session(self):
+        ''' Setup Boto3 session '''
+        session = Session(aws_access_key_id=self.access_key_id,
+                          aws_secret_access_key=self.secret_access_key)
+
+        if not self.account_id:
+            self.account_id = session.client('sts').get_caller_identity().get('Account')
+            self.credentials[self.account_id] = (self.access_key_id, self.secret_access_key, None)
+
+        self.sessions[self.account_id] = session
+        self.sqs_res.pop(self.account_id, None)
+
+    def _setup_multiaccount_session(self, account_id, cross_roles_arns):
+        ''' Assume role and setup session for the cross-account capability'''
+        try:
+            assumed_role = self.sessions[self.account_id].client('sts').assume_role(
+                RoleArn=cross_roles_arns[account_id],
+                RoleSessionName='StackStormEvents'
+            )
+        except ClientError:
+            self._logger.error('Could not assume role on %s', account_id)
+            return
+
+        self.credentials[account_id] = (assumed_role["Credentials"]["AccessKeyId"],
+                                        assumed_role["Credentials"]["SecretAccessKey"],
+                                        assumed_role["Credentials"]["SessionToken"])
+
+        session = Session(
+            aws_access_key_id=self.credentials[account_id][0],
+            aws_secret_access_key=self.credentials[account_id][1],
+            aws_session_token=self.credentials[account_id][2]
+        )
+        self.sessions[account_id] = session
+        self.sqs_res.pop(account_id, None)
+
+    def _setup_sqs(self, session, account_id, region):
+        ''' Setup SQS resources'''
+        if region in self.sqs_res[account_id]:
+            return self.sqs_res[account_id][region]
 
         try:
-            self.sqs_res = self.session.resource('sqs')
+            self.sqs_res[account_id][region] = session.resource('sqs', region_name=region)
+            return self.sqs_res[account_id][region]
         except NoRegionError:
-            self._logger.warning("The specified region '%s' is invalid", self.aws_region)
+            self._logger.error("The specified region '%s' for account %s is invalid.",
+                               region, account_id)
+
+    def _get_info(self, queue):
+        ''' Retrieve the account ID and region from the queue URL '''
+        # Pull default values from previous configuration
+        account_id = self.account_id
+        aws_region = self.aws_region
+
+        # Netloc will be empty if the queue is just a name
+        if queue.netloc:
+            try:
+                account_id = queue.path.split('/')[1]
+            except IndexError as e:
+                six.reraise(type(e), type(e)(
+                    "Queue URL must contain the account ID as the first part of the path, "
+                    "eg: https://.../<account_id>"),
```

Comment:
Now I'm thinking we should be even more explicit:
```suggestion
                    "eg: https://sqs.<aws_region>.amazonaws.com/<account_id>/<queue_name>"),
```